webpack/localization-plugin/src/LocalizationPlugin.ts

```diff
@@ -0,0 +1,732 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import { JsonFile } from '@microsoft/node-core-library';
+import * as Webpack from 'webpack';
+import * as path from 'path';
+import * as lodash from 'lodash';
+import * as Tapable from 'tapable';
+
+/**
+ * @public
+ */
+export interface ILocJsonFileData {
+  [stringName: string]: string;
+}
+
+/**
+ * @public
+ */
+export interface ILocale {
+  [locJsonFilePath: string]: ILocJsonFileData;
+}
+
+/**
+ * @public
+ */
+export interface ILocales {
+  [locale: string]: ILocale;
+}
+
+/**
+ * @public
+ */
+export interface IDefaultLocaleOptions {
+  locale?: string;
+  usePassthroughLocale?: boolean;
+
+  /**
+   * If {@link IDefaultLocaleOptions.usePassthroughLocale} is set, use this name for the passthrough locale.
+   * Defaults to "passthrough"
+   */
+  passthroughLocaleName?: string;
+}
+
+/**
+ * The options for localization.
+ *
+ * @public
+ */
+export interface ILocalizationPluginOptions {
+  localizedStrings: ILocales;
+  defaultLocale: IDefaultLocaleOptions;
+  serveLocale: IDefaultLocaleOptions;
+  filesToIgnore?: string[];
+  localizationStatsDropPath?: string;
+  localizationStatsCallback?: (stats: ILocalizationStats) => void;
+}
+
+/**
+ * @internal
+ */
+export interface IStringPlaceholder {
+  value: string;
+  suffix: string;
+}
+
+const PLUGIN_NAME: string = 'localization';
+const LOCALE_FILENAME_PLACEHOLDER: string = '[locale]';
+const LOCALE_FILENAME_PLACEHOLDER_REGEX: RegExp = new RegExp(lodash.escapeRegExp(LOCALE_FILENAME_PLACEHOLDER), 'g');
+const STRING_PLACEHOLDER_PREFIX: string = '-LOCALIZED-STRING-f12dy0i7-n4bo-dqwj-39gf-sasqehjmihz9';
```

Comment:
Did you intend for this to start with a dash `-`?

Reply:
Yeah.